modules/dnn/CMakeLists.txt

```diff
@@ -131,11 +131,11 @@ endif()
 
 if(HAVE_TENGINE)
 	list(APPEND include_dirs ${TENGINE_INCLUDE_DIRS})
-	if(EXISTS ${TENGINE_LIBRARIES})
-		list(APPEND libs ${TENGINE_LIBRARIES})
+    if(EXISTS ${TENGINE_LIBRARIES})
+		list(APPEND libs -Wl,--whole-archive ${TENGINE_LIBRARIES} -Wl,--no-whole-archive)
```

Comment:
> if(EXISTS ${TENGINE_LIBRARIES})

Don't work with list as variable value.
Broken indentation.

No difference between both code paths. `ocv_add_dependencies()` is misused (required opencv modules) and does nothing.

Reply:
done.